protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -66,149 +67,31 @@
-          // if not, using the same MethodDescriptor would ensure the parser matches and permit us
-          // to enable this optimization.
-          if (protoStream.parser() == parser) {
-            try {
-              @SuppressWarnings("unchecked")
-              T message = (T) ((ProtoInputStream) stream).message();
-              return message;
-            } catch (IllegalStateException ex) {
-              // Stream must have been read from, which is a strange state. Since the point of this
-              // optimization is to be transparent, instead of throwing an error we'll continue,
-              // even though it seems likely there's a bug.
-            }
-          }
-        }
-        CodedInputStream cis = null;
-        try {
-          if (stream instanceof KnownLength) {
-            int size = stream.available();
-            if (size > 0 && size <= DEFAULT_MAX_MESSAGE_SIZE) {
-              // buf should not be used after this method has returned.
-              byte[] buf = bufs.get().get();
-              if (buf == null || buf.length < size) {
-                buf = new byte[size];
-                bufs.set(new WeakReference<byte[]>(buf));
-              }
-
-              int remaining = size;
-              while (remaining > 0) {
-                int position = size - remaining;
-                int count = stream.read(buf, position, remaining);
-                if (count == -1) {
-                  break;
-                }
-                remaining -= count;
-              }
-
-              if (remaining != 0) {
-                int position = size - remaining;
-                throw new RuntimeException("size inaccurate: " + size + " != " + position);
-              }
-              cis = CodedInputStream.newInstance(buf, 0, size);
-            } else if (size == 0) {
-              return defaultInstance;
-            }
-          }
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-        if (cis == null) {
-          cis = CodedInputStream.newInstance(stream);
-        }
-        // Pre-create the CodedInputStream so that we can remove the size limit restriction
-        // when parsing.
-        cis.setSizeLimit(Integer.MAX_VALUE);
-
-        try {
-          return parseFrom(cis);
-        } catch (InvalidProtocolBufferException ipbe) {
-          throw Status.INTERNAL.withDescription("Invalid protobuf byte sequence")
-            .withCause(ipbe).asRuntimeException();
-        }
-      }
-
-      private T parseFrom(CodedInputStream stream) throws InvalidProtocolBufferException {
-        T message = parser.parseFrom(stream, globalRegistry);
-        try {
-          stream.checkLastTagWas(0);
-          return message;
-        } catch (InvalidProtocolBufferException e) {
-          e.setUnfinishedMessage(message);
-          throw e;
-        }
-      }
-    };
+    return new ProtobufLiteMessageMarshaller<T>(defaultInstance);
```

Comment:
nit: The class name can be shorter like `LiteMessageMarshaller` or even `MessageMarshaller`, as this is a private inner class, ProtobufLite is presumed. The same for `MetadataMarshaller`.

Reply:
Done.